src/ProductConstructionService/ProductConstructionService.DependencyFlow/PullRequestConflictNotifier.cs

```diff
@@ -0,0 +1,170 @@
+    /// </summary>
+    Task NotifyAboutConflictingUpdateAsync(
+        ConflictInPrBranchException conflictException,
+        SubscriptionUpdateWorkItem update,
+        Subscription subscription,
+        InProgressPullRequest pr);
+
+    /// <summary>
+    /// Posts a comment in the PR when the PR has a conflict with the target branch.
+    /// This means, there might be conflicts in the version files and the comment will
+    /// provide guidance on how to resolve those.
+    /// </summary>
+    Task NotifyAboutMergeConflictAsync(
+        InProgressPullRequest pr,
+        SubscriptionUpdateWorkItem update,
+        Subscription subscription,
+        IReadOnlyCollection<UnixPath> conflictedFiles);
+}
+
+internal class PullRequestConflictNotifier : IPullRequestConflictNotifier
+{
+    private readonly IRemoteFactory _remoteFactory;
+    private readonly ILogger<PullRequestConflictNotifier> _logger;
+
+    public PullRequestConflictNotifier(
+        IRemoteFactory remoteFactory,
+        ILogger<PullRequestConflictNotifier> logger)
+    {
+        _remoteFactory = remoteFactory;
+        _logger = logger;
+    }
+
+    /// <summary>
+    /// Handles a case when new code flow updates cannot be flowed into an existing PR,
+    /// because the PR contains a change conflicting with the new updates.
+    /// In this case, we post a comment on the PR with the list of files that are in conflict,
+    /// </summary>
+    public async Task NotifyAboutConflictingUpdateAsync(
+        ConflictInPrBranchException conflictException,
+        SubscriptionUpdateWorkItem update,
+        Subscription subscription,
+        InProgressPullRequest pr)
+    {
+        // The PR we're trying to update has a conflict with the source repo. We will mark it as blocked, not allowing any updates from this
+        // subscription till it's merged, or the conflict resolved. We'll set a reminder to check on it.
+        StringBuilder sb = new();
+        sb.AppendLine($"There was a conflict in the PR branch when flowing source from {update.GetRepoAtCommitUri()}");
+        sb.AppendLine("Files conflicting with the head branch:");
+        foreach (var file in conflictException.FilesInConflict)
+        {
+            sb.AppendLine($" - [{file}]({update.GetFileAtCommitUri(file)})");
+        }
+        sb.AppendLine();
+        sb.AppendLine("Updates from this subscription will be blocked until the conflict is resolved, or the PR is merged");
+
+        var remote = await _remoteFactory.CreateRemoteAsync(subscription.TargetRepository);
+
+        try
+        {
+            await remote.CommentPullRequestAsync(pr.Url, sb.ToString());
+        }
+        catch (Exception e)
+        {
+            _logger.LogWarning("Posting comment to {prUrl} failed with exception {message}", pr.Url, e.Message);
+        }
+    }
+
+    /// <summary>
+    /// Handles a case when new code flow updates cannot be flowed into an existing PR,
```

Comment:
the description is wrong here, it's the same as in `NotifyAboutConflictingUpdateAsync`